opentelemetry/proto/trace/v1/trace.proto

```diff
@@ -109,6 +109,16 @@ message Span {
   // field must be empty. The ID is an 8-byte array.
   bytes parent_span_id = 4;
 
+  // Flags, a bit field. 8 least significant bits are the trace
+  // flags as defined in W3C Trace Context specification. Readers
+  // MUST not assume that 24 most significant bits will be zero.
+  // When creating new spans, the most-significant 24-bits MUST be
+  // zero.  To read the 8-bit W3C trace flag (use flags &
+  // SPAN_FLAGS_TRACE_FLAGS_MASK).  [Optional].
```

Comment:
I think "when creating new spans" may still be a bit confusing in the protocol spec as it could be understood as "creating new span messages", especially since the remainder seems to assume that "new spans" have a 32 bit flags field. I could not find a shorter unambiguous way than this:

```suggestion
  // MUST not assume that 24 most significant bits will be zero.
  // To read the 8-bit W3C trace flag (use flags & SPAN_FLAGS_TRACE_FLAGS_MASK).
  // When creating span messages, if the message is logically forwarded from another source
  // with an equivalent flags fields (i.e., usually another OTLP span message), the field SHOULD
  // be copied as-is. If creating from a source that does not have an equivalent flags field
  // (such as a runtime representation of an OpenTelemetry span), the high 24 bits MUST
  // be set to zero.
  // [Optional].
```

Reply:
Also, there seems to be a typo in:

> To read the 8-bit W3C trace flag (use flags & SPAN_FLAGS_TRACE_FLAGS_MASK).

I assumed it should mean:

> To read the 8-bit W3C trace flag, use `flags & SPAN_FLAGS_TRACE_FLAGS_MASK`.

Reply:
I think we forgot to apply an analogous change to the flags field in the link object.